### PR TITLE
Allow fine tuning of StoreQueryParameters

### PR DESCRIPTION
**Pull request by sobychacko**
When using InteractiveQueryService in Kafka Streams binder, it is requested to customize the StoreQueryParameters beyond what is done through the InteractiveQueryService itself. Provide a StoreQueryParametersCustomizer that enables this customization.

Resolves https://github.com/spring-cloud/spring-cloud-stream/issues/2608

### CODE REVIEW

File: binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/InteractiveQueryService.java [hunk 1]
```diff
@@ -298,10 +306,17 @@ public <K> KafkaStreams getKafkaStreams(String store, K key, Serializer<K> seria
 	public List<HostInfo> getAllHostsInfo(String store) {
 		return kafkaStreamsRegistry.getKafkaStreams()
 				.stream()
-				.flatMap(k -> k.allMetadataForStore(store).stream())
+				.flatMap(k -> k.streamsMetadataForStore(store).stream())
```
Comment (onobc) [inline binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/InteractiveQueryService.java:309]: How is this change related?

Comment (sobychacko) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/InteractiveQueryService.java:309]: It is unrelated; I just saw that the original call was deprecated and decided to clean up inline.

Comment (onobc) [reply binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/InteractiveQueryService.java:309]: I would leave it as-is for this PR. The reason being is that it is deprecated in kafka-streams 3.3.1 but I think there is slight behavior change in the 2 methods across earlier versions of kafka-streams.

**Review by onobc — APPROVED** [review]
@sobychacko LGTM.
